custom_components/kumo/config_flow.py

```diff
@@ -103,7 +190,40 @@ async def async_step_init(self, user_input=None):
                 vol.Required("response_timeout", default=8): str,
             }
         )
-        return self.async_show_form(step_id="init", data_schema=data_schema)
+
+        return self.async_show_form(step_id="timeout_settings", data_schema=data_schema)
+
+    async def async_step_unit_select(self, user_input=None):
+        """Handle options flow."""
+
+        kumo_cache = await self.hass.async_add_executor_job(
+            load_json, self.hass.config.path(KUMO_CONFIG_CACHE)
+        )
+        kumo_unit_list = {}
+        for child in kumo_cache[2]["children"]:
+            for raw_unit in child["zoneTable"].values():
+                kumo_unit_list[str(raw_unit["label"])] = (
+                    str(raw_unit.get("address", "empty")),
+                )
```

Comment:
...and here, and in the next loop. It may be worth a pair of utility functions to get / set the IPs for all the units